### ISSUE TITLE
/etc is writable with persistent and shared content

### ISSUE THREAD

**Issue by bam80** [flatpak/flatpak#2299]
@TingPing @alexlarsson 
## Linux distribution and version
Slackware 14.2
## Flatpak version
1.0.4
built with:
https://slackbuilds.org/slackbuilds/14.2/desktop/flatpak/flatpak.SlackBuild
## Description of the problem
/etc/ content is writable, persistent and shared among sandboxes using the same Platform.
Actually, the Platform itself is freely modifiable from within the sandboxes this way.
## Steps to reproduce

- buid/install sample app, e.g. _org.flatpak.Hello_: http://docs.flatpak.org/en/latest/first-build.html
(I was installed Platform as user)
- `$ flatpak run --command=sh org.flatpak.Hello`
- note _/etc/profile.d_ is writable:
```
$ mount|grep profile.d
/dev/sdb1 on /etc/profile.d type ext4 (rw,nosuid,nodev,relatime,data=ordered)
bam@darkstar:~$ touch /etc/profile.d/a
bam@darkstar:~$ ls -l /etc/profile.d/
total 8
-rw-r--r-- 1 bam bam    0 Nov 11 18:37 a
-rw-r--r-- 3 bam bam 1107 Jan  1  1970 gawk.csh
-rw-r--r-- 3 bam bam  757 Jan  1  1970 gawk.sh
```

**Comment by bam80** [comment]
I'm ready to debug this but since I'm newbie to Flatpak I need some assistance

**Comment by kirbyfan64** [comment]
Some more stuff of interest:

- This only works with user instances. System ones result in *Permission denied*...
- ...because this actually modifies the OSTree checkout at `~/.local/share/flatpak/.../files/etc/profile.d`. You can even overwrite and modify the file. Can't this break the fact that everything is supposed to be OSTree-managed?

**Comment by bam80** [comment]
> Writing anything to /etc gets (correctly) discarded.

Yes, because /etc itself is just a dir on / rw tmpfs

**Comment by TingPing** [comment]
As I mentioned on IRC at a glance it could be as simple as this line should be `ro-bind`:

https://github.com/flatpak/flatpak/blob/master/common/flatpak-run.c#L2469

**Comment by matthiasclasen** [comment]
this was fixed in 1.0.5
